### ISSUE TITLE
feature: custom openai base url

### ISSUE THREAD

**Issue by ImSingee** [windmill-labs/windmill#4572]
Now windmill supports OPENAI_AZURE_BASE_PATH to use azure instead of OpenAI. 

But I may want to use other third-party solutions, like openrouter or cloudflare ai gateway. 

Azure uses a different way to pass the API key compared to openai. But other providers use the same way as OpenAI. So just use their endpoint as OPENAI_AZURE_BASE_PATH doesn't work.

**Comment by rubenfiszel** [comment]
Just a note that the fact OPENAI_AZURE_BASE_PATH is currently an oversight considering in the UI it's marked as an ee feature. We will later refactor the ai integration to allow for more public ais but we will also restrict private and local models to enterprise and remove OPENAI_AZURE_BASE_PATH for being available in non ee.

**Comment by cabusar** [comment]
Dear Ruben,

"we will also restrict private and local models to enterprise"

Like almost every options in Windmill ?

Don't get me wrong, your product is amazing and i'm totally ok for you to make money with it, in particular when it's used by enterprises.

But open source didn't mean "publish the code somewhere where everyone could read it and the braves would use it as some sort of freemium application if they manage to install it in a reliable way". In fact your own product is full of open source parts and you should be glad none of them has "enterprise features" because you wouldn't had the chance to launch Windmill then.

In this particular case, you want to restrict the use of custom AI to EE, thinking probably only big companies have custom models and the rest of us have nothing able to run AI models.

Guess what ? You totally wrong, many people use smaller models on high end consumer GPU, especially when they try do dev things before going on AI providers, it work enough to do amazing things.

It's ok, your product is restrictive enough regarding what is "free" and what is not, no company will ever dare to install the community edition or use it in production, risks are already to high. 
That's sad because i know many sysadmin that use to test the community edition to check if everything match their expectations before trying to sell it to management. And no, the one month free is not enough for them and the "contact sales" will not do because internal processes are rigids and wouldn't allow them to contact a sales without involving at least the boss of their boss and god know who.

Hope you will hear this and try to target "enterprise features" without harming the hobbyist/local lab guys that could promote the product or help you find bugs/securities issues.

**Comment by HugoCasa** [comment]
@cabusar "Like almost every options in Windmill ?" really?

anyway, the custom AI provider is part of the OSS/CE edition since [v1.479.0](https://github.com/windmill-labs/windmill/releases/tag/v1.479.0)

**Comment by cabusar** [comment]
Yeah really, in instance configuration, 22 options out of 28 are enterprise only.

In addition there is more in every corner of windmill.
![Image](https://github.com/user-attachments/assets/d3cf623c-d30c-4cd9-9113-13657673d563)
![Image](https://github.com/user-attachments/assets/46a5dfb8-348d-42cb-9b18-c51f5ce9af03)
![Image](https://github.com/user-attachments/assets/60b63614-5615-401d-be1a-4eade6ec5779)
![Image](https://github.com/user-attachments/assets/aa86ad77-5611-425b-8049-a4c7565ab16d)
![Image](https://github.com/user-attachments/assets/57f2f24e-4c92-4718-bd39-bad2e3d73368)
![Image](https://github.com/user-attachments/assets/e26e11dc-22fc-43b4-8904-24ef547b625a)
![Image](https://github.com/user-attachments/assets/894ac9b0-aaf2-42af-8c75-366dd943526e)
![Image](https://github.com/user-attachments/assets/ff014ad5-d0b5-402b-8478-e5f692d01ed7)
![Image](https://github.com/user-attachments/assets/954e16be-2b4c-4006-a163-39c445c207b7)
![Image](https://github.com/user-attachments/assets/f437f766-ae9c-4426-9d6d-dbd40f5c7210)
![Image](https://github.com/user-attachments/assets/062d1e20-f84d-4933-9f7e-a873f3fbf193)
![Image](https://github.com/user-attachments/assets/ab1076ba-8ed7-4717-b9fe-9514a61654f1)
![Image](https://github.com/user-attachments/assets/84f79594-755f-4150-8e46-cb0b4e4dd4f9)
![Image](https://github.com/user-attachments/assets/d8d7cc5c-11c4-42f2-b8d2-26c45c04378c)
![Image](https://github.com/user-attachments/assets/b67ece14-f5b8-4a03-8299-403b73b2dd0b)
![Image](https://github.com/user-attachments/assets/7b325286-0530-4adb-bdf5-0acf53575361)

So, once again don't get me wrong, Windmill is really great, you have every right to make money out of it, and i'm glad there is a free version that i can use to test the product but don't play dumb when a user tell you that's not what open source mean.

Anyway, i digress from the original topic and as OPENAI_AZURE_BASE_PATH is tagged EE and the custom AI provider will not be an EE feature, if i understand you well, i think you should close this issue as it answer the original request for feature.

Have a nice day.
